Mp metrics2x

evelinec opened the pull request:
Update guide to mpMetrics-2.0 according to issues raised in https://github.com/OpenLiberty/guides-common/issues/224, #50. 

Specific changes: 
- Updated metric outputs where the formats are different, ie, `@Gauged`, `@Counted` and some base metric outputs
- Added Liberty vendor metrics
- Updated `@Countered` annotation description to be counting monotonically upwards only
- Added tags for the `@Timed` annotation to demonstrate multi-dimension metrics
- Added description of the multi-dimension metrics with tag and updated guide accordingly
- Updated the `start` directory

gkwan-ibm commented:
- [ ] suggest the description of `@Timed` of `get()`
description = "Time needed to get the properties of a system"
- [ ] suggest the description of `@Timed` of `list()`
description = "Time needed to create an inventory list of systems"

- [ ] the output from 90 to 97 can be better as
```
# TYPE application_inventoryProcessingTime_rate_per_second gauge
application_inventoryProcessingTime_rate_per_second{method="get"} 0.0019189661542898407
...
# TYPE application_inventoryProcessingTime_seconds summary
# HELP application_inventoryProcessingTime_seconds Time needed to get the properties of a system
application_inventoryProcessingTime_seconds_count{method="get"} 1
application_inventoryProcessingTime_seconds{method="get",quantile="0.5"} 0.127965469
...
# TYPE application_inventoryProcessingTime_rate_per_second gauge
application_inventoryProcessingTime_rate_per_second{method="list"} 0.0038379320982686884
...
# TYPE application_inventoryProcessingTime_seconds summary
# HELP application_inventoryProcessingTime_seconds Time needed to create an inventory list of systems
application_inventoryProcessingTime_seconds_count{method="list"} 2
application_inventoryProcessingTime_seconds{method="list",quantile="0.5"} 2.2185000000000002E-5
...
```

evelinec commented:
Addressed review feedback. We need to have the same name and description for the tagged annotation to work. We can't change the descriptions as suggested, unless they're the same. 

@gkwan-ibm can you review again? Thank you.

evelinec commented:
@Channyboy Hi David, can you help SME review the changes make to update the metrics guide to mpMetrics-2.0? Detail changes described in the PR description. Thank you.

Channyboy commented on README.adoc:
```diff
@@ -116,9 +129,26 @@ base:jvm_uptime_seconds 39.025
 ----
 [role="no_copy"]
 ----
```
The base metrics here should be `base_*` not `base:*`

evelinec replied:
Fixed.

Channyboy commented on README.adoc:
```diff
@@ -178,34 +206,61 @@ configuration to log in to view the data.
 
-Apply the [hotspot=42-44]`@Counted` annotation to the [hotspot=45-47]`list()` method to count how many times the
-http://localhost:9080/inventory/systems[http://localhost:9080/inventory/systems^] URL is accessed. 
+Both the [hotspot=get]`get()` and [hotspot=list]`list()` methods are annotated with the `@Timed` metric and have the same [hotspot=nameForGet hotspot=nameForList]`inventoryProcessingTime` name. The `@Timed` annotation tracks how frequently and how long it takes for the method invocation to complete. The [hotspot=tagForGet]`method=get` and [hotspot=tagForList]`method=list` tags add a dimension to  ...
```
The first sentence indicates that the metrics use the same name. Maybe the second sentence should discuss the use of tags which then adds the extra dimension of unique identification for the metrics?

Channyboy commented on README.adoc:
```diff
@@ -178,34 +206,61 @@ configuration to log in to view the data.
+Both the [hotspot=get]`get()` and [hotspot=list]`list()` methods are annotated with the `@Timed` metric and have the same [hotspot=nameForGet hotspot=nameForList]`inventoryProcessingTime` name. The `@Timed` annotation tracks how frequently and how long it takes for the method invocation to complete. The [hotspot=tagForGet]`method=get` and [hotspot=tagForList]`method=list` tags add a dimension to  ...
+
+- The `method=get` processing time tag indicates the time to get the system properties from calling the `system` service.
+- The `method=list` processing time tag indicates the time for the `inventory` service to list all of the system properties in the inventory.
```
Sounds a little bit weird to me, sounds like we're saying the tag itself is indicating time?

evelinec replied:
Fixed

evelinec commented on README.adoc:
```diff
@@ -178,34 +208,62 @@ configuration to log in to view the data.
+You can see the vendor only metrics in the `metrics/vendor` endpoint.
+You see metrics from the runtime components, such as Web Application, ThreadPool and Session Management.
+Note that these metrics are specific to Liberty application server. Different vendor may provide other metrics.
+Visit https://www.ibm.com/support/knowledgecenter/en/SSD28V_liberty/com.ibm.websphere.wlp.core.doc/ae/rwlp_monitor_metrics_rest_api.html[Liberty vendor metrics^] for more information.
```
@Charlotte-Holt The suggested linked doc doesn't have immediate references to the topic in this section, ie, Liberty vendor metrics.

Charlotte-Holt replied:
@evelinec When we created this topic I believe that we came to the conclusion that the differentiation between vendor, base, etc. wouldn't be information that most users would be concerned with. @donbourne Do you think that we should include anything in the [Metrics reference list](https://openliberty.io/docs/ref/general/#metrics-catalog.html) to call out vendor-specific metrics for the purposes of this guide?

donbourne replied:
we are planning to add the prometheus names to that reference list ( see https://github.com/OpenLiberty/open-liberty/issues/8822 ).  Those names include a prefix that shows which of base / vendor the metrics come from.  @evelinec , will that address what you need?

evelinec commented:
@yeekangc Can you review this PR as well pls? Thank you.

evelinec commented:
Updated Liberty vendor doc link to the online doc, as https://github.com/OpenLiberty/docs/issues/684 is being worked on already.

evelinec commented:
Merged to `QA`.
